### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Ruslan-Mykhalchak**
[DEMO LINK](https://Ruslan-Mykhalchak.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by Mariia-Torkaienko — APPROVED** [review]
Good job!

File: src/components/CurrentUser/CurrentUser.tsx [hunk 1]
```diff
@@ -1,12 +1,68 @@
...
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.userId !== this.props.userId) {
+      this.loadUser();
+    }
+  }
+
+  loadUser = () => {
+    getUsers(this.props.userId)
```
Comment (aliceblunt8) [inline src/components/CurrentUser/CurrentUser.tsx:30]:
```suggestion
    getUser(this.props.userId)
```

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,44 +1,130 @@
...
+  filterBySelect = () => {
+    if (this.state.selected === 'active') {
+      return this.filterByTitle().filter(todo => todo.completed !== true);
+    }
+
+    if (this.state.selected === 'completed') {
+      return this.filterByTitle().filter(todo => todo.completed === true);
+    }
+
+    return this.filterByTitle();
```
Comment (aliceblunt8) [inline src/components/TodoList/TodoList.tsx:44]: it's better to use switch case in this case

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -1,44 +1,130 @@
...
+    });
+  };
+
+  handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ selected: event.target.value });
+  };
+
+  filterBySelect = () => {
+    if (this.state.selected === 'active') {
+      return this.filterByTitle().filter(todo => todo.completed !== true);
```
Comment (aliceblunt8) [inline src/components/TodoList/TodoList.tsx:37]:
```suggestion
      return this.filterByTitle().filter(todo => !todo.completed);
```

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,44 +1,130 @@
...
+    }
+
+    if (this.state.selected === 'completed') {
+      return this.filterByTitle().filter(todo => todo.completed === true);
```
Comment (aliceblunt8) [inline src/components/TodoList/TodoList.tsx:41]:
```suggestion
      return this.filterByTitle().filter(todo => todo.completed);
```

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,44 +1,130 @@
...
+    }
+
+    return this.filterByTitle();
+  };
+
+  render() {
+    const { currentUser, selectedUserId } = this.props;
+    const { query } = this.state;
+
+    const filteredTodos = this.filterBySelect();
```
Comment (aliceblunt8) [inline src/components/TodoList/TodoList.tsx:51]:
```suggestion
    const filteredTodos = this.getfilteredTodos();
```

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**